activesupport/lib/active_support/file_update_checker.rb

```diff
@@ -44,7 +44,7 @@ def initialize(files, dirs = {}, &block)
         raise ArgumentError, "A block is required to initialize a FileUpdateChecker"
       end
 
-      @files = files.freeze
+      @files = files.dup.freeze
```

Comment:
The `freeze` was added at 5ea2b11ad7fb1e0e0580edd9d8e0e688cfffa3f2 to ensure the files will not be changed.

Generally passed argument should not be mutated, but https://github.com/Shopify/bootsnap/issues/106 is strange to me. If `can't modify frozen Array` is raised by the `freeze`, it means that the files are changed after FileUpdateChecker has been initialized, it should not be happenned for FileUpdateChecker. Isn't it?